packages/hapi/modules/plugin/plugin.ts

```diff
@@ -99,33 +125,39 @@ const createPlugin = ({ options: pluginOptions } = {}) => {
         if (doesResponseNeedInvocation) response.continue();
       };
 
-      // NOTE: This version detection allows us to graceully support
-      // both new and old Hapi APIs.
+      // NOTE: This version detection allows us to graceully support both new and old Hapi APIs.
+      // This is very hard to type as we would have to import two aliased versions of types.
       if (areServerEventsSupported) {
+        // @ts-ignore
         server.ext('onRequest', onRequestHandler);
+        // @ts-ignore
         server.events.on('response', onResponseHandler);
       } else {
+        // @ts-ignore
         server.ext('onRequest', onRequestHandler);
+        // @ts-ignore
         server.ext('onPreResponse', onResponseHandler);
       }
 
-      server.decorate('server', 'Prometheus', Prometheus);
+      server.decorate('server', 'Prometheus', () => Prometheus);
```

Comment:
This seems to be a breaking change, no?

Reply:
You might be right about that. Sorry that slipped my eyes. I can not recall why I changes it. You can open a PR to reverse it if you want.

Reply:
Thanks, meanwhile I've opened a separate issue about it: https://github.com/tdeekens/promster/issues/372